src/ray/raylet/node_manager.cc

```diff
@@ -3050,6 +3051,36 @@ void NodeManager::HandleNodeStatsRequest(const rpc::GetNodeStatsRequest &request
   }
 }
 
+void NodeManager::HandleProfilingStatsRequest(const rpc::GetProfilingStatsRequest &request,
```

Comment:
Why not handle this stats request in job agent or some plugin like it?
It's heavy overhead in raylet/nodemanager.

Reply:
This is now fixed btw (it goes through the reporter, which is off the critical path). Thanks for the suggestion :)